src/MobSpawner.cpp

```diff
@@ -208,12 +208,27 @@ bool cMobSpawner::CanSpawnHere(cChunk * a_Chunk, Vector3i a_RelPos, eMonsterType
 		case mtMagmaCube:
 		case mtSlime:
 		{
+			auto maxLight = Random.RandInt(0, 7);
+			auto phaseNumber = int(std::floor(a_Chunk->GetWorld()->GetWorldAge().count() / 24000)) % 8;
 			return
 			(
 				(TargetBlock == E_BLOCK_AIR) &&
 				(BlockAbove == E_BLOCK_AIR) &&
-				((!cBlockInfo::IsTransparent(BlockBelow)) || (a_DisableSolidBelowCheck)) &&
-				((a_RelPos.y <= 40) || (a_Biome == biSwampland))
+				(
+					(!cBlockInfo::IsTransparent(BlockBelow)) ||
+					(a_DisableSolidBelowCheck)) &&
+				(
+					(a_RelPos.y <= 40) ||
+					(
+						(a_Biome == biSwampland) &&
+						(a_RelPos.y >= 50) &&
+						(a_RelPos.y <= 70) &&
+						(SkyLight <= maxLight) &&
+						(BlockLight <= maxLight) &&
+						((std::abs(phaseNumber - 4) / 4) > Random.RandReal()) &&
+						(Random.RandBool(0.5))
+					)
+				)
```

Comment:
If I understood `If the fraction of the moon that is bright is greater than a random number (from 0 to 1)`  correctly I would expect something like this:
```cpp
const float AMOUNT_MOON_PHASES = 8f;
auto moonPhase = static_cast<float>(std::floor(a_Chunk->GetWorld()->GetWorldAge().count() / 24000)) % AMOUNT_MOON_PHASES;
auto threshold = std::abs(static_cast<float>(moonPhase) - (AMOUNT_MOON_PHASES / 2)) / (AMOUNT_MOON_PHASES / 2);

if (GetRandomProvider().RandBool(threshold))
{
	// .. 
}
```

Also, I'm not really sure why `maxLight` is a random value. It seems a slime can spawn at any lightlevel under 7, just not every tick.

Reply:
Is the behaviour in the PR not equivalent to the code you have suggested? It only differs in using `threshold > Random.RandReal()` rather than `Random.RandBool(threshold)`

maxLight being a random value is what's specified in the wiki so unless that's wrong no need to differ. It makes sense that slimes should spawn more often at lower light levels in any case.

Reply:
Ah yes, it's documented in the wiki. Just ignore this comment.